### PR TITLE
Deploy temp v1.9 ERC20Factory

### PR DESCRIPTION
**Pull request by wilsoncusack**
Deployed on [Base](https://basescan.org/address/0xf10122d428b4bc8a9d050d06a2037259b4c4b83b) and [Base Sepolia](https://sepolia.basescan.org/address/0xf10122d428b4bc8a9d050d06a2037259b4c4b83b) at address 0xf10122d428b4bc8a9d050d06a2037259b4c4b83b. 

We will need to wait for #158 in order to get this functionality at `0x4200000000000000000000000000000000000012`, but as our docs are already pointing to a abnormal address (due to a past address collision issue from `create`), we can update our docs immediately with this new address.

### CODE REVIEW

**Comment by wilsoncusack** [comment]
Note that [Sepolia canonical address](https://sepolia.basescan.org/address/0x4200000000000000000000000000000000000012#writeProxyContract) already has a version that supports decimals, but wanted to deploy anyway to allow devs 1:1 testing.

File: mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "forge-std/Script.sol";
+import {ERC20Factory} from "../src/ERC20Factory.sol";
+
+contract RunERC20FactoryDeploy is Script {
+    function run() public {
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
```
Comment (anikaraghu) [inline mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol:10]: could we use a K2 key?

File: mainnet/2024-04-30-deployTempERC20Factory/deployed/addresses.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
[the remark this answers is not in the document]
Comment (wilsoncusack) [reply mainnet/2024-04-30-deployTempERC20Factory/deployed/addresses.json:1]: What are the cases to use one vs. the other? I just followed the README

File: mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol (hunk 1 above)
Comment (wilsoncusack) [reply mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol:10]: Why?

File: mainnet/2024-04-30-deployTempERC20Factory/deployed/addresses.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (anikaraghu) [reply mainnet/2024-04-30-deployTempERC20Factory/deployed/addresses.json:1]: hm yeah could probably be more clear/named better in the README, but deploy template is typically used to deploy an entirely new network (eg all OP contracts), generic template can be used for one-off tasks like a deployment of a single contract or an upgrade etc.

File: mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol (hunk 1 above)
Comment (anikaraghu) [reply mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol:10]: we've generally avoided any production level interaction with local keys (see assessment [here](https://docs.google.com/document/d/1zGJdXvIESQvg3hcryFrUmnTl6tJnvsn8v3oPVDq7wdY/edit#heading=h.q2wnua8mdrtn))

Comment (wilsoncusack) [reply mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol:10]: What's the paved road for this? This doesn't seem to be documented here?

File: mainnet/2024-04-30-deployTempERC20Factory/deployed/addresses.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (wilsoncusack) [reply mainnet/2024-04-30-deployTempERC20Factory/deployed/addresses.json:1]: oh I see, what's the `make` command for that? I can remove empty folders/files

File: mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        vm.startBroadcast(deployerPrivateKey);
+        new ERC20Factory{salt: '0xBA5ED'}();
```
Comment (nadir-akhtar-coinbase) [inline mainnet/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol:12]: Shouldn't impact functionality in this case but better practice, especially if someone copies this code pattern down the line w/o knowing why there is no `stopBroadcast()`
```suggestion
        new ERC20Factory{salt: '0xBA5ED'}();
        vm.stopBroadcast();
```

File: sepolia/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+pragma solidity 0.8.15;
+
+import "forge-std/Script.sol";
+import {ERC20Factory} from "../src/ERC20Factory.sol";
+
+contract RunERC20FactoryDeploy is Script {
+    function run() public {
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        vm.startBroadcast(deployerPrivateKey);
+        new ERC20Factory{salt: '0xBA5ED'}();
```
Comment (nadir-akhtar-coinbase) [inline sepolia/2024-04-30-deployTempERC20Factory/script/RunDeployERC20Factory.sol:12]: Same as with the other deploy script
```suggestion
        new ERC20Factory{salt: '0xBA5ED'}();
        vm.stopBroadacast();
```

**Review by nadir-akhtar-coinbase — COMMENTED** [review]
@wilsoncusack Small remarks, but otherwise looks good. Also, would be nice as Anika mentioned to remove excess files for clarity and ease of review

**Review by nadir-akhtar-coinbase — APPROVED** [review]
Accepting nits staying unresolved in favor of making progress on publicizing the decimal-supporting factory

**Pull request merged**